### PR TITLE
[MAIN-2169] Prometheus preformance fix

### PR DESCRIPTION
**Pull request by Avi-Robusta**
We updated prometrix for faster preformance.
https://pypi.org/project/prometrix/0.2.0/

There were issues with speed and cpu when serialising the Hikaru object.
you can see the speed improvements [here](https://github.com/robusta-dev/prometrix/pull/25)

As you can see it is used practically in the same way but now needs some slight changes
- dict(block.data) to de-serialise it 
- the sub objects from PrometheusQueryResult are all treated as a dict object 
   - i.e instead of ``query_result.vector_result[0].value`` we need to use ``query_result.vector_result[0]["value"]``
   
###   Tests ran
-   UI - (krr, nodes, metrics exporter, apps page + apps custom graphs)
-   DB - (verified the node average usage query reports correct data to DB)
- Actions - actions impacted - prometheus_sla_enricher - graph enrichers like node_graph_enricher

### CODE REVIEW

File: src/robusta/core/sinks/robusta/prometheus_discovery_utils.py [hunk 1]
```diff
@@ -60,7 +60,7 @@ def _get_query_prometheus_value(self, query: str) -> Optional[float]:
             if query_result.result_type == "error" or not query_result.vector_result:
                 logging.error("PrometheusDiscoveryUtils failed to get prometheus results.")
                 return
-            value = query_result.vector_result[0].value.value
+            value = query_result.vector_result[0]["value"]["value"]
```
Comment (Avi-Robusta) [inline src/robusta/core/sinks/robusta/prometheus_discovery_utils.py:63]: I verified this works/is correct in the DB

File: src/robusta/core/sinks/robusta/dal/model_conversion.py [hunk 2]
```diff
@@ -154,7 +154,7 @@ def to_evidence_json(
                 structured_data.append({"type": "list", "data": block.items})
             elif isinstance(block, PrometheusBlock):
                 structured_data.append(
-                    {"type": "prometheus", "data": block.data.dict(), "metadata": block.metadata, "version": 1.0}
+                    {"type": "prometheus", "data": dict(block.data), "metadata": block.metadata, "version": 1.0}
```
Comment (Avi-Robusta) [inline src/robusta/core/sinks/robusta/dal/model_conversion.py:157]: Verified this works via frontend

File: pyproject.toml [hunk 3]
```diff
@@ -60,7 +60,7 @@ humanize = "^3.13.1"
 cryptography = "^42.0.5"
 fpdf2 = "^2.7.1"
 attrs = "^23.1.0"
-prometrix = "0.1.16"
+prometrix = "0.1.19a3"
```
Comment (Avi-Robusta) [inline pyproject.toml:63]: needs to be updated to next version of prometrix

File: src/robusta/core/playbooks/prometheus_enrichment_utils.py [hunk 4]
```diff
@@ -114,13 +114,13 @@ def get_resolution_from_duration(duration: timedelta) -> int:
 
 def get_target_name(series: PrometheusSeries) -> Optional[str]:
     for label in ["container", "pod", "node"]:
-        if label in series.metric:
-            return series.metric[label]
+        if label in series["metric"]:
```
Comment (arikalon1) [inline src/robusta/core/playbooks/prometheus_enrichment_utils.py:117]: Why do we need this?
`PrometheusSeries` isn't a dict
It does have a field named `metric`

File: src/robusta/core/playbooks/prometheus_enrichment_utils.py [hunk 5]
```diff
@@ -114,13 +114,13 @@ def get_resolution_from_duration(duration: timedelta) -> int:
...
+            return series["metric"][label]
     return None
 
 
 def get_series_job(series: PrometheusSeries) -> Optional[str]:
-    return series.metric.get("job")
+    return series["metric"]["job"] if "job" in series["metric"] else None
```
Comment (arikalon1) [inline src/robusta/core/playbooks/prometheus_enrichment_utils.py:123]: same

File: src/robusta/core/playbooks/prometheus_enrichment_utils.py [hunk 6]
```diff
@@ -209,21 +209,21 @@ def create_chart_from_prometheus_query(
         label = get_target_name(series)
 
         if not label:
-            label = "\n".join([v for (key, v) in series.metric.items() if key != "job"])
+            label = "\n".join([v for (key, v) in series["metric"].items() if key != "job"])
```
Comment (arikalon1) [inline src/robusta/core/playbooks/prometheus_enrichment_utils.py:212]: I'm not sure how this works
The `series_list_result` contains 2 different data types
On line 204, it gets a `List[dict]` , and on line 206 a `List[PrometheusSeries]`
PrometheusSeries isn't a dict, so how is this working?

File: src/robusta/core/playbooks/prometheus_enrichment_utils.py [hunk 7]
```diff
@@ -209,21 +209,21 @@ def create_chart_from_prometheus_query(
...
 
         # If the label is empty, try to take it from the additional_label_factory
         if label == "" and chart_label_factory is not None:
             label = chart_label_factory(i)
 
         values = []
-        for index in range(len(series.values)):
-            timestamp = series.timestamps[index]
-            value = round(float(series.values[index]), FLOAT_PRECISION_LIMIT)
+        for index in range(len(series["values"])):
```
Comment (arikalon1) [inline src/robusta/core/playbooks/prometheus_enrichment_utils.py:219]: I think the type of `series` is ambiguous
It's a `dict` or `PrometheusSeries`

**Review by arikalon1 — COMMENTED** [review]
nice work

left some comments

File: src/robusta/core/playbooks/prometheus_enrichment_utils.py (hunk 4 above)
Comment (Avi-Robusta) [reply src/robusta/core/playbooks/prometheus_enrichment_utils.py:117]: typing is unclear it is a dict, fixing

File: src/robusta/core/playbooks/prometheus_enrichment_utils.py (hunk 5 above)
Comment (Avi-Robusta) [reply src/robusta/core/playbooks/prometheus_enrichment_utils.py:123]: typing is unclear it is a dict, fixing

File: src/robusta/core/playbooks/prometheus_enrichment_utils.py (hunk 6 above)
Comment (Avi-Robusta) [reply src/robusta/core/playbooks/prometheus_enrichment_utils.py:212]: typing is unclear it is a dict, fixing

File: src/robusta/core/playbooks/prometheus_enrichment_utils.py (hunk 7 above)
Comment (Avi-Robusta) [reply src/robusta/core/playbooks/prometheus_enrichment_utils.py:219]: typing is unclear it is a dict, fixing

**Review by arikalon1 — APPROVED** [review]
nice work

**Pull request merged**